lib/stdlib/src/io_ansi.erl

```diff
@@ -0,0 +1,2099 @@
+1> io_ansi:cursor_backward().
+<<"\b">>
+```
+""".
+?SPEC(cursor_backward).
+?FUNCTION(cursor_backward).
+
+-doc """
+Move the cursor backward `N` characters.
+
+Example:
+```erlang
+1> io_ansi:cursor_backward(42).
+<<"\e[42D">>
+```
+""".
+?SPEC(cursor_backward, N).
+?FUNCTION(cursor_backward, N).
+
+-doc """
+Move the cursor to the start of the current line.
+
+Example:
+```erlang
+1> io_ansi:cursor_home().
+<<"\e[H">>
+```
+""".
+?SPEC(cursor_home).
+?FUNCTION(cursor_home).
+
+-doc """
+Move the cursor up one line, but keeps the cursor on the same location on the
+screen by scrolling the screen down.
+
+Example:
+```erlang
+1> io_ansi:reverse_index().
+<<"\eM">>
+```
+""".
+?SPEC(reverse_index).
+?FUNCTION(reverse_index).
+
+-doc """
+Save the current cursor position.
+
+Example:
+```erlang
+1> io_ansi:cursor_save().
+<<"\e7">>
+```
+""".
+?SPEC(cursor_save).
+?FUNCTION(cursor_save).
+
+-doc """
+Restore a saved cursor position.
+
+Example:
+```erlang
+1> io_ansi:cursor_restore().
+<<"\e8">>
+```
+""".
+?SPEC(cursor_restore).
+?FUNCTION(cursor_restore).
+
+-doc """
+Show the cursor.
+
+Example:
+```erlang
+1> io_ansi:cursor_show().
+<<"\e[?12;25h">>
+```
+""".
+?SPEC(cursor_show).
+?FUNCTION(cursor_show).
+
+-doc """
+Hide the cursor.
+
+Example:
+```erlang
+1> io_ansi:cursor_hide().
+<<"\e[?25l">>
+```
+""".
+?SPEC(cursor_hide).
+?FUNCTION(cursor_hide).
+
+-doc """
+Move the cursor down one line and then returns it to home.
+
+Example:
+```erlang
+1> io_ansi:cursor_next_line().
+<<"\eE">>
+```
+""".
+?SPEC(cursor_next_line).
+?FUNCTION(cursor_next_line).
+
+-doc """
+Move the cursor up one line and then returns it to home.
+
+Example:
+```erlang
+1> io_ansi:cursor_previous_line().
+<<"\e[F">>
+```
+""".
+?SPEC(cursor_previous_line).
+?FUNCTION(cursor_previous_line).
+
+-doc """
+Move the cursor to column `X`.
+
+Example:
+```erlang
+1> io_ansi:cursor_horizontal_absolute(10).
+<<"\e[11G">>
+```
+""".
+?SPEC(cursor_horizontal_absolute, X).
+?FUNCTION(cursor_horizontal_absolute, X).
+
+-doc """
+Move the cursor to line `X`.
+
+Example:
+```erlang
+1> io_ansi:cursor_vertical_absolute(20).
+<<"\e[21d">>
+```
+""".
+?SPEC(cursor_vertical_absolute, X).
+?FUNCTION(cursor_vertical_absolute, X).
+
+-doc """
+Move the cursor to line `X` and column `Y`.
+
+Example:
+```erlang
+1> io_ansi:cursor_horizontal_vertical(10, 20).
+<<"\e[10;20f">>
+```
+""".
+?SPEC(cursor_horizontal_vertical, X, Y).
+?FUNCTION(cursor_horizontal_vertical, X, Y).
+
+-doc """
+Instruct the terminal to report the current cursor position.
+
+Examples:
+
+```erlang
+1> io_ansi:cursor_report_position().
+~"\e[6n"
+```
+
+```bash
+## Enter noshell-raw mode and request curson location and then print
+## the reply to stdout.
+$ erl -noshell -eval 'shell:start_interactive({noshell,raw}),
+    io_ansi:fwrite([cursor_report_position]),
+    io:format("~p",[io:get_chars("",20)])' -s init stop
+"\e[58;1R"
+```
+""".
+?SPEC(cursor_report_position).
+?FUNCTION(cursor_report_position).
```

Comment:
Is a `cursor_get_position() -> {Col:integer(), Row:integer()}`  feasible? 
So I don't have to parse the ANSI codes, when debugging? 
Or do I do that with one of the terminfo functions?